### ISSUE TITLE
Investigate round trip routing

### ISSUE THREAD

**Issue by rabidllama** [GIScience/openrouteservice#391]
Graphhopper core offers round trip routing which could be useful if implemented in ors for things such as hikes, jogging, bike trips etc.

We need to check to see if it is 
1) Functional in GH
2) How to implement it in ors
3) Decide on API calls
4) Make it use the various restrictions available within ors

**Comment by samcrawford** [comment]
I have a significant interest in this feature. I can vouch for the fact that it works in GH, I tested it only recently (it does not work in their web UI though).

The approach used is fairly simple and I've been toying with making a few improvements. In short, the current approach picks N waypoints, on a random initial bearing, distributed evenly in a circle from the start point up to the supplied target distance. It then plots a route using these waypoints.

The downside to this is that the initial point selection is entirely arbitrary, which can lead to weird routes being plotted (especially when rivers are involved), and to the final route's distance being significantly larger than the target distance. This is most noticeable with shorter target distances (<10km).

Additionally, I have been wondering if functionality for "create me a route between A and B up to a desired target distance" would be more useful. A generic "target distance" feature would still allow round-trip routes to be calculated (A and B are the same). But it would also allow you to go from point A to B, whilst extending the route to ~X km. This is useful for people targeting a specific distance for walking/cycling/jogging.

I hope to work on this "target distance" functionality myself soon, and would be happy to contribute it, but I'm only just getting started with ORS so don't hold your breath!

**Comment by rabidllama** [comment]
Hi @samcrawford 

Thanks for that, I had a look the other day integrating the round trip stuff into the ORS API and it seemed to be a bit hit and miss. As you said, for shorter distances I ended up getting much larger distances than I requested (walking set to 21km ended up sometimes coming back with a 31km route).

I think the main things that we would need to ensure happens is that for these shorter routes, the results are still somewhat in the region of what was requested, and that the avoidables could also be taken into account (one thought is to add higher weightings to traffic lights/crossings for pedestrians as if you want the route for jogging, having to stop and wait at those can be a major pain).

The other functionality you suggested could certainly be interesting, but I think that it could be quite a difficult thing to accomplish, especially if we want to do it in a way that is fast for the request and I think that would be something that has to be modified in the Graphhopper code to then be implemented in ORS. But certainly something worth looking into.
